Classes/KIFUITestActor.m

```diff
@@ -229,7 +229,7 @@ - (void)waitForAnimationsToFinishWithTimeout:(NSTimeInterval)timeout stabilizati
             [window performBlockOnDescendentViews:^(UIView *view, BOOL *stop) {
                 BOOL isViewVisible = [view isVisibleInViewHierarchy];   // do not wait for animations of views that aren't visible
                 BOOL hasUnfinishedSystemAnimation = [NSStringFromClass(view.class) isEqualToString:@"_UIParallaxDimmingView"];  // indicates that the view-hierarchy is in an in-between-state of an animation
-                if (isViewVisible && ([view.layer hasAnimations] || hasUnfinishedSystemAnimation)) {
```

Comment:
If you take a look at CALayer-KIFAdditions.m, the `hasAnimations` method is more complicated than just checking the number of `animationKeys`. If there is something wrong with the `hasAnimations` method, it should probably be patched there.